data/topic-filter.go

```diff
@@ -0,0 +1,222 @@
+package data
+
+import (
+	"context"
+	"net/url"
+	"strings"
+
+	errs "github.com/ONSdigital/dp-frontend-search-controller/apperrors"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// Filter represents information of filters selected by user
+type TopicFilter struct {
```

Comment:
TopicFilter object is the same as Filter struct - so no need for TopicFilter struct just use Filter struct (use inheritance). The Filter struct does have a generic name so it shouldn't be confusing